packages/repo-metadata-lint/src/validate.ts

```diff
@@ -31,20 +33,25 @@ const API_LIBRARY_TYPES = [
   'GAPIC_COMBO',
 ];
 
-interface ApiIndex {
-  apis: {hostName: string}[];
-}
-
-const GOOGLEAPIS_OWNER = 'googleapis';
-const GOOGLEAPIS_REPO = 'googleapis';
-const GOOGLEAPIS_DEFAULT_BRANCH = 'master';
-
 // Apply validation logic to .repo-metadata.json.
 export class Validate {
-  octokit: Octokit;
-  cachedApiIndex?: ApiIndex;
-  constructor(octokit: Octokit) {
-    this.octokit = octokit;
+  allowedApiShortnames: Set<string>;
+  constructor(allowedApiShortnames: Set<string>) {
+    this.allowedApiShortnames = allowedApiShortnames;
+  }
+  static async build(
```

Comment:
Why not add these arguments to the constructor?

Reply:
I'm not a huge fan of this class reaching out and doing stuff implicitly given an octokit or GCS connection/bucket.

The class really only cares about the allowed shortnames and the schema. Therefore, I opted to put the API calls into a helper method and pass only the data necessary for the validator into the class.